internal/testing/env.go

```diff
@@ -0,0 +1,127 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package testing
+
+import (
+	"os"
+)
+
+type Env struct {
+	Name   string
+	Value  string
+	Exists bool
+}
+
+// EnvStore stores and recovers environment variables.
+type EnvStore interface {
+	Add(env Env)
+
+	Remove(key string)
+
+	Reset()
+
+	// Records the environment variable into the store.
+	Record(key string)
+
+	// Restore recover the environment variables in the store.
+	Restore() error
+
+	// Setenv sets the value of the environment variable named by the key and
+	// records original environment variable.
+	// It returns an error, if any.
+	Setenv(key, value string) error
+}
```

Comment:
Can we simplify the interface?

   1. For example, I don't think `Add()/Remove()/Reset()` need to be part of the exported interface - but could be unexported functions of the concrete `envStore`. They don't seem to be called anywhere in the code that uses this interface.

   2. What is the distinction between `Record(key)` and `Setenv(key, value)`.  From the perspective of the user (developer in this case), what is the difference between calling `Setenv(key, val)` vs. a `Record(key) followed by `os.SetEnv(key, val)` (or `os.UnsetEnv(key)` for that matter). 

   3. Could we get away with just `Record(key)` that the developer calls for each environment variable they know they *may* be modifying, and then a `Restore()` in the deferred func.

Reply:
Yeah, I think this interface is overdesign. I can simplify this and remain `Record` and `Restore`.